src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php

```diff
@@ -112,21 +112,24 @@ public function onKernelResponse(FilterResponseEvent $event)
     protected function injectToolbar(Response $response, Request $request)
     {
         $content = $response->getContent();
-        $pos = strripos($content, '</body>');
-
-        if (false !== $pos) {
-            $toolbar = "\n".str_replace("\n", '', $this->twig->render(
-                '@WebProfiler/Profiler/toolbar_js.html.twig',
-                array(
-                    'position' => $this->position,
-                    'excluded_ajax_paths' => $this->excludedAjaxPaths,
-                    'token' => $response->headers->get('X-Debug-Token'),
-                    'request' => $request,
-                )
-            ))."\n";
-            $content = substr($content, 0, $pos).$toolbar.substr($content, $pos);
-            $response->setContent($content);
+        $injectToolbarAt = strripos($content, '</body>') ?: strripos($content, '</html>');
+        $canInjectToolbar = $injectToolbarAt || stripos($content, '<title>');
```

Comment:
something including ``</body>`` or ``</html>`` is safe, as we want to inject things at the end of the page, and the detection precisely relies on identifying the injection place. The issue IMO is checking for ``<title>`` to detect that we should inject it, and then injecting at the end of the response because we cannot detect the right injection place.